content/en/docs/concepts/services-networking/dns-pod-service.md

```diff
@@ -232,10 +232,10 @@ following Pod-specific DNS policies. These policies are specified in the
   stub-domain and upstream DNS servers configured.
   See [related discussion](/docs/tasks/administer-cluster/dns-custom-nameservers)
   for details on how DNS queries are handled in those cases.
-- "`ClusterFirstWithHostNet`": For Pods running with hostNetwork, you should
-  explicitly set its DNS policy to "`ClusterFirstWithHostNet`". Otherwise, Pods
-  running with hostNetwork and `"ClusterFirst"` will fallback to the behavior
-  of the `"Default"` policy.
+- "`ClusterFirstWithHostNet`": For Pods running with hostNetwork, effectively
+  uses "`ClusterFirst`" policy if kubelet was configured with a Cluster DNS option.
+  Otherwise if kubelet Cluster DNS was not configured, falls back to "`Default`"
+  (see above).
```

Comment:
I don't think that's accurate? Well, and "a Cluster DNS option" is not clear...

This whole section needs to be rewritten... eg, the fact that `ClusterFirst` falls back to `Default` for hostNetwork pods should be mentioned under `ClusterFirst`, not under `ClusterFirstWithHostNet`.

(Also we should totally rename all of the options and deprecate the old names...)

Reply:
Ugh, all possible configurations of `dnsPolicy` and `hostNetwork` involve some amount of "it depends on how kubelet and host-level DNS are configured".

I think in _this_ document we should just be assuming that everything is configured "correctly". Unfortunately, it is not clear to me what that is.

At any rate, the description of `ClusterFirst` should say that it falls back to `Default` for host-network pods, and the description of `ClusterFirstWithHostNet` should probably copy most of the text from `ClusterFirst` and clarify that it means "always behave the same way whether or not the pod is host-network".

It should definitely not be claiming that anyone "should" do anything, because that depends both on how the cluster is configured and on what their desired result is.